### DISCUSSION TITLE
Is it possible to modify application inner function, for instance browser...

### DISCUSSION THREAD

**Discussion by KnIfER** [ramensoftware/windhawk#190] [Q&A]
Yesterday I've learned how to intercept and modify windows system calls from certain application. 

Now I wonder is it possible to modify inner functions of app ?

 For instance the chrome browser have many limitations, some page can't be shown inside of a iframe, is  it possible to unleash that hard-coded limitation, if I've found the source code?

what tool do I need, Frida?

**Comment by m417z** [comment]
Yes, it's certainly possible. Just a couple of days ago, a mod was published which does something like this with Chrome: [Chrome UI Tweaks](https://windhawk.net/mods/chrome-ui-tweaks).
Regarding tools, I like using x64dbg, but other tools like Frida or WinDbg might work too, depending on your needs and preferences.

**Comment by KnIfER** [comment]
Thanks! I will learn how to use x64dbg first. 

Are there any tutorials on how to get those binary instructions used in **Chrome UI Tweaks**?  
![image](https://github.com/ramensoftware/windhawk/assets/11593903/338bdccb-86c4-4523-b46c-04f768ea11e3)

I found one step-by-step tutorial (in chinese) : [chrome extension developer mode warning remover-CSDN](https://blog.csdn.net/tian_ci/article/details/90046995) .   Steps are : load chrome.dll , press alt+f9 multiple times until run into that dll module, and then right-click and search for string references..

Another tutorial (which I cannot understand) : [Hooking Chrome’s SSL functions – Nytro Security](https://nytrosecurity.wordpress.com/2018/02/26/hooking-chromes-ssl-functions/)

But still, dont know how to locate the binary instructions of specific function .

**Reply by m417z** [reply]
There are many tutorials, it's difficult to just name one. Look for x64dbg and reverse engineering tutorials.
